source/rpt/parameters_rpt.cc

```diff
@@ -202,7 +200,56 @@ Parameters::DetectorParameters::parse_parameters(ParameterHandler &prm)
     radius                  = prm.get_double("radius");
     length                  = prm.get_double("length");
     detector_positions_file = prm.get("detector positions file");
+
+    // Transform string to vector of double
+    std::string dead_time_string  = prm.get("dead time");
```

Comment:
See how this is done here : 
https://github.com/lethe-cfd/lethe/blob/cda6095c0c3f2df9c2e0fb9e1dfca967d350536d/source/core/parameters_lagrangian.cc#L363

Use the same approach, it will make the code simpler and less error prone (and also more readable :))